### PR TITLE
HepMC2 recipe

### PR DESCRIPTION
**Pull request by gartung**
Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages
- [x] Build number is 0
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

File: recipes/hepmc2/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    - patches/hepmc-2.06.07-WeightContainer-fix-size_type.patch
+    - patches/HepMC-2.06.07-nodoc.patch
+build:
+  number: 0
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - {{ compiler('fortran') }}
```
[the remark this answers is not in the document]
Comment (gartung) [reply recipes/hepmc2/meta.yaml]: I was going by the CMS RPM spec which sets $F77.

[the remark this answers is not in the document]
Comment (gartung) [reply recipes/hepmc2/meta.yaml]: There is one Fortran file in the example/fio directory but it is not compiled.

File: recipes/hepmc2/test.sh [hunk 2]
```diff
@@ -0,0 +1,11 @@
+#!/usr/bin/env bash
```
Comment (chrisburr) [inline recipes/hepmc2/test.sh]: Renaming this to `run_test.sh` will cause it to be ran implicitly (or `run_test.py`/`run_test.bat`)

Comment (gartung) [reply recipes/hepmc2/test.sh]: The error is reported as 
line 7 command not found

Comment (chrisburr) [reply recipes/hepmc2/test.sh]: That makes sense as it's not in `PATH`, running it explicitly would need `${RECIPE_DIR}/test.sh`

**Comment by gartung** [comment]
So there are 2 tests failing. The test scripts are comparing two text files after running sed on them. This seems a little error prone. I tried running the cmake build instead and running ctest. This also fails because they are calling the same scripts. I am not sure it is critical to have those tests pass.

File: recipes/hepmc2/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    - {{ compiler('cxx') }}
+    - cmake
+
+about:
+  home: http://hepmc.web.cern.ch/hepmc/
+  license: LGPL-2.1
+  license_family: LGPL
+  summary: 'HepMC - a C++ Event Record for Monte Carlo Generators'
+
+  # The remaining entries in this section are optional, but recommended
```
Comment (chrisburr) [inline recipes/hepmc2/meta.yaml]:
```suggestion
```

File: recipes/hepmc2/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    - source/
+  requires:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - cmake
+
+about:
+  home: http://hepmc.web.cern.ch/hepmc/
+  license: LGPL-2.1
+  license_family: LGPL
```
Comment (chrisburr) [inline recipes/hepmc2/meta.yaml]: As most open source licences require a copy of the licence to be included this field should always be included:
```suggestion
  license_family: LGPL
  license_file: source/INSTALL
```
(tinyxml2 only needed a copy in the recipe directory as it's not included in their source tarball)

File: recipes/hepmc2/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+            (Comput. Phys. Commun. 134 (2001) 41). Since the first 
+           implementation, the package has undergone several modifications
+           and in particular, the latest HepMC3 series is a completely new
+           re-write using currently available C++11 techniques. 
+
+extra:
+  recipe-maintainers:
+    - gartung
+    - chrisburr
+    - henryiii
```
Comment (chrisburr) [inline recipes/hepmc2/meta.yaml]: @henryiii Are you happy to be listed as a maintainer of this?

File: recipes/hepmc2/run_test.sh [hunk 6]
```diff
@@ -0,0 +1,7 @@
+#!/usr/bin/env bash
```
Comment (chrisburr) [inline recipes/hepmc2/run_test.sh]: This file should be removed as it doesn't do anything.

File: recipes/hepmc2/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+    - patches/HepMC-2.06.07-broken-tests.patch
+build:
+  number: 0
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - cmake
+  host:
```
Comment (chrisburr) [inline recipes/hepmc2/meta.yaml]:
```suggestion
```

File: recipes/hepmc2/meta.yaml [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+
+  run:
```
Comment (chrisburr) [inline recipes/hepmc2/meta.yaml]:
```suggestion
```

File: recipes/hepmc2/meta.yaml [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+
+test:
+  source_files:
+    - source/
```
Comment (chrisburr) [inline recipes/hepmc2/meta.yaml]:
```suggestion
    - source/examples
```

File: recipes/hepmc2/meta.yaml [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+  requires:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - cmake
```
Comment (chrisburr) [inline recipes/hepmc2/meta.yaml]: It's nice to have some kind of test to make sure the package actually works after being relocated. How about using this for testing?
```suggestion
  commands:
    # Workaround for lack of CONDA_BUILD_SYSROOT variable in staged recipes
    - export CONDA_BUILD_SYSROOT=/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.9.sdk  # [osx]
    - cd source/examples
    # example_UsingIterators
    - ${CXX} example_UsingIterators.cc -o example_UsingIterators -I${CONDA_PREFIX}/include -lhepmc
    # Fix up the runpath on macOS
    - ${INSTALL_NAME_TOOL} -change @rpath/libHepMC.4.dylib $CONDA_PREFIX/lib/libHepMC.4.dylib example_UsingIterators # [osx]
    - ./example_UsingIterators
    # example_EventSelection
    - ${CXX} example_EventSelection.cc -o example_EventSelection -I${CONDA_PREFIX}/include -lhepmc
    # Fix up the runpath on macOS
    - ${INSTALL_NAME_TOOL} -change @rpath/libHepMC.4.dylib $CONDA_PREFIX/lib/libHepMC.4.dylib example_EventSelection # [osx]
    - ./example_EventSelection
    # example_BuildEventFromScratch
    - ${CXX} example_BuildEventFromScratch.cc -o example_BuildEventFromScratch -I${CONDA_PREFIX}/include -lhepmc
    # Fix up the runpath on macOS
    - ${INSTALL_NAME_TOOL} -change @rpath/libHepMC.4.dylib $CONDA_PREFIX/lib/libHepMC.4.dylib example_BuildEventFromScratch # [osx]
    - ./example_BuildEventFromScratch
```

File: recipes/hepmc2/meta.yaml [hunk 11]
```diff
@@ -0,0 +1,57 @@
...
+  url: http://lcgapp.cern.ch/project/simu/HepMC/download/HepMC-{{ version }}.tar.gz
+  sha256: a0bdd6f36a3cc4cb59d6eb15cef9d46ce9b3739cae3324e81ebb2df6943e4594
+  folder: source
+  patches:
+    - patches/hepmc-2.03.06-reflex.patch
+    - patches/hepmc-2.06.07-WeightContainer-fix-size_type.patch
+    - patches/HepMC-2.06.07-nodoc.patch
+    - patches/HepMC-2.06.07-broken-tests.patch
+build:
+  number: 0
```
Comment (chrisburr) [inline recipes/hepmc2/meta.yaml]: There should either be a `bld.bat` script for windows or it should be skipped. At the moment the windows build is an empty package.
```suggestion
  number: 0
  skip: True  # [win]
```

File: recipes/hepmc2/meta.yaml (hunk 5 above)
Comment (henryiii) [reply recipes/hepmc2/meta.yaml]: Sure.

**Comment by henryiii** [comment]
You should be able to add gnu sed on macOS. Add ‘sed’ to the test requirements.

**Comment by gartung** [comment]
I already had to disable three tests using the autotools generated shell scripts. The output or floats appears to be incorrectly formatted and comparing to what look like ints fails. There first four columns of number agree. Its just the last two columns. If you read the comments in the srcipts there were already tests diabled under cygwin because the formatting was broken.

**Comment by gartung** [comment]
I would rather move on to the Fireworks build.

**Pull request merged**